### PR TITLE
Added firebird_finance dex

### PR DESCRIPTION
**Pull request by ARDev097**
Added trades data for firebird_finance_dex using macro approach.

### CODE REVIEW

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ARDev097

**Comment by Hosuke** [comment]
By the way, please add the `firebird_finance` info in this file:
https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_info.sql

(Maybe also add the `mummy_finance` please)

**Review by Hosuke — CHANGES_REQUESTED** [review]
By the way, please add the `firebird_finance` info in this file:
https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_info.sql

(Maybe also add the `mummy_finance` please)

File: models/dex/dex_trades.sql [hunk 1]
```diff
@@ -78,6 +78,7 @@
 ,ref('woofi_optimism_trades')
 ,ref('oneinch_lop_own_trades')
 ,ref('mummy_finance_optimism_trades')
+,ref('firebird_finance_optimism_trades')
```
Comment (Hosuke) [inline models/dex/dex_trades.sql:81]: Do you think we can put `firebird_finance_optimism_trades` into `dex_aggregator.trades`?

**Comment by ARDev097** [comment]
Hello @Hosuke,

I have made the required changes as suggested in the PR. Please resolve these conflicts so that these checks can be completed. Let me know if any further changes are required from my end.

File: models/_sector/dex/trades/optimism/dex_optimism_base_trades.sql [hunk 2]
```diff
@@ -11,6 +11,7 @@
     , ref('mummy_finance_optimism_base_trades')
     , ref('sushiswap_v1_optimism_base_trades')
     , ref('sushiswap_v2_optimism_base_trades')
+    , ref('firebird_finance_optimism_base_trades')
```
Comment (Hosuke) [inline models/_sector/dex/trades/optimism/dex_optimism_base_trades.sql:14]: If `firebird_finance_optimism_base_trades` is under `dex_aggregator.trades`, then we can remove it from `dex.base_trades`. (As currently dex.base_trades should only contain trades in dex.trades)

File: seeds/dex/trades/dex_trades_seed.csv [hunk 3]
```diff
@@ -529,4 +529,7 @@ avalanche_c,uniswap,3,2023-11-11,0x31e1a422f13f742ba7ad4c5b2b33bf2cc6c05777cfff4
...
-optimism,mummy_finance,1,2023-06-09,0xd7cadb14f39a79bf0ca3ea0ebf9aa85caa1465da154010294ee502942a8de5d4,21,0x4200000000000000000000000000000000000006,1.530834771081591,0x4200000000000000000000000000000000000042,2097.160814236567
\ No newline at end of file
+optimism,mummy_finance,1,2023-06-09,0xd7cadb14f39a79bf0ca3ea0ebf9aa85caa1465da154010294ee502942a8de5d4,21,0x4200000000000000000000000000000000000006,1.530834771081591,0x4200000000000000000000000000000000000042,2097.160814236567
+optimism,firebird_finance,1,2023-06-21,0xee6c8e576530a282dd8b6b0150690d2efc9bafcf535b994a9efdd95c68d6aaf0,31,0x7f5c764cbc14f9669b88837ca1490cca17c31607,28.088612,0x47536f17f4ff30e64a96a7555826b8f9e66ec468,18.109264993439986
+optimism,firebird_finance,1,2022-09-12,0x4b229101a53eda64d7205e8c67da5b69ac57b7030d7420777131dcebd1b6451c,18,0x4200000000000000000000000000000000000006,0.01913919808305519,0x7f5c764cbc14f9669b88837ca1490cca17c31607,33.450938
+optimism,firebird_finance,1,2022-09-27,0x77266a3fb99476be6859436eb0000ec1d0a0a78400149195317a65e2a70a5577,11,0x7f5c764cbc14f9669b88837ca1490cca17c31607,400,0x4200000000000000000000000000000000000042,413.41739538888197
```
Comment (Hosuke) [inline seeds/dex/trades/dex_trades_seed.csv:535]: Also you can move those seed records into this file:
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/aggregator/dex_aggregator_seed.csv

**Review by Hosuke — CHANGES_REQUESTED** [review]

**Comment by ARDev097** [comment]
Hello @Hosuke ,
I have made the requested changes; please check them.

**Comment by Hosuke** [comment]
I will remove `firebird_finance` from dex.base_trades first.

File: models/firebird_finance/optimism/firebird_finance_optimism_schema.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - blockchain
+            - project
+            - version
+            - tx_hash
+            - evt_index 
+      - check_dex_seed:
```
Comment (Hosuke) [inline models/firebird_finance/optimism/firebird_finance_optimism_schema.yml:23]:
```suggestion
      - check_dex_aggregator_seed:
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ARDev097 

[Sample firebird_finance_optimism_trades]()

**Review by jeff-dude — APPROVED** [review]
thanks for the collab here, lgtm 🤝

**Pull request merged**